python/ray/data/impl/plan.py

```diff
@@ -186,15 +202,16 @@ def _rewrite_read_stage(self) -> Tuple[BlockList, "Stage"]:
         """
         # Generate the "GetReadTasks" stage blocks.
         remote_args = self._in_blocks._read_remote_args
-        blocks = []
+        read_tasks = []
         metadata = []
+        # TODO(Clark): Add CreationStage so we're not shoe-horning read tasks into a
+        # BlockList.
```

Comment:
Can you remove this?

Reply:
If we're able to find a way to not put raw `ReadTasks` (non-object references) into a `BlockList`, which is a very hacky use of `BlockList`, I'll remove this TODO.

Reply:
I think you can store the base read tasks in the plan class instead of in the block list.

Reply:
Yep that's what I had imagine this `CreationStage` to be, basically an alternative source for the execution plan that doesn't require input blocks.